### ISSUE TITLE
Ability to add pre-planned routined schedule delivery in Task Scheduler

### ISSUE THREAD

**Issue by razilzul** [open-rmf/rmf-web#380]
### Status

Blocked by open-rmf/rmf_ros2#103.

### Description

*Adding this feature request here based on Morgan's advice.*

Have additional options for task creation to specify tasks that need to be repeated, or start on a condition other than time. The task overview should also be able to reflect these details as well as allow for editing.

### Implementation considerations

'Task->Task Scheduler' additional options should include:
- At a specific time
- At a specific time on a daily schedule
- At a specific time on a weekly schedule
- At a specific time on a monthly schedule
- At a specific time on a monthly day-of-week schedule
- *OPTIONAL* When a specific event occurs (i.e. start, end of another task)

Scheduled task overview list 'Task->Overview' should include:
- The schedule of the task should also be indicated
- Allow for edits to change the schedule
- Allow for enabling/disabling
- Deleting tasks

### Additional information
Could emulate existing task scheduling interfaces or alternatively look at clock (alarm) apps on the phone for the execution of the implementation flow.

**Comment by matiasbavera** [comment]
@codebot this is just a mock, but I think we cover all the cases with the following UI (except `When a specific event occurs (i.e. start, end of another task)`) .

![Peek 2021-07-05 19-32](https://user-images.githubusercontent.com/11761240/124524152-e54a1500-ddc7-11eb-87cc-f0e15db329bb.gif)

Any feedback is welcome =)

**Comment by aaronchongth** [comment]
This can be achieved using the task scheduler now
